compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -839,9 +840,33 @@ public void visit(BLangArrayLiteral arrayLiteral) {
     }
 
     public void visit(BLangRecordLiteral recordLiteral) {
-        recordLiteral.keyValuePairs.forEach(kv -> {
+        List<BLangRecordLiteral.BLangRecordKeyValue> keyValuePairs = recordLiteral.keyValuePairs;
+        keyValuePairs.forEach(kv -> {
             analyzeExpr(kv.valueExpr);
         });
+
+        Set<Object> names = new TreeSet<>((l, r) -> l.equals(r) ? 0 : 1);
+        for (BLangRecordLiteral.BLangRecordKeyValue recFieldDecl : keyValuePairs) {
+            BLangExpression key = recFieldDecl.getKey();
+            if (key.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                BLangSimpleVarRef keyRef = (BLangSimpleVarRef) key;
+                if (names.contains(keyRef.variableName.value)) {
+                    logDuplicateKeyError(recordLiteral, keyRef, key.pos);
+                }
+                names.add(keyRef.variableName.value);
+            } else if (key.getKind() == NodeKind.LITERAL) {
+                BLangLiteral keyLiteral = (BLangLiteral) key;
+                if (names.contains(keyLiteral.value)) {
+                    logDuplicateKeyError(recordLiteral, keyLiteral, key.pos);
+                }
+            }
+        }
+    }
+
+    private void logDuplicateKeyError(BLangRecordLiteral recordLiteral, BLangExpression key, DiagnosticPos pos) {
```

Comment:
Do we really need this method? We can inline this code segment very easily which will only add a single line above and then we can remove this method completely.